crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -125,5 +137,11 @@ def addon_installed?(addon)
     rescue NameError
       false
     end
+
+    def repo_version_available(product, version)
+      `zypper products`.split("\n").detect do |p|
```

Comment:
This will pick only the first ocurence of product. But as the old product was still not reinstalled, there will be both shown by zypper output:
```
LANG=C zypper products
S | Repository | Internal Name        | Name                                              | Version | Arch   | Is Base
--+------------+----------------------+---------------------------------------------------+---------+--------+--------
i | @System    | SLES                 | SUSE Linux Enterprise Server 12 SP1               | 12.1-0  | x86_64 | Yes    
i | @System    | suse-openstack-cloud | SUSE OpenStack Cloud 6                            | 6-0     | x86_64 | No     
v | Cloud      | suse-openstack-cloud | SUSE OpenStack Cloud 7                            | 7-0     | x86_64 | No     
  | SDK-SP2    | sle-sdk              | SUSE Linux Enterprise Software Development Kit 12 | 12.2-0  | x86_64 | No     
v | sles12sp2  | SLES                 | SUSE Linux Enterprise Server 12 SP2 RC2           | 12.2-0  | x86_64 | No 
```

Good test case for your unit tests...